### PR TITLE
Added Path PostProcessor for old WEDM machines

### PR DESCRIPTION
**Pull request by alromh87**
Added Path PostProcessor for old WEDM machines configurable for different machines:

	G91 incremental coordinates
		--relative				(Dissabled by default)
	Specific comment character
		--comments-character			(None by default)
	Specific space character, use "" to ommit spaces
		--command-space				(" " by default)
	Add character before newline
		--endline-character			(None by default)
	Multiplier for machines that use not standard dimensions like micrometers (1000)
		--scale					(1 by default)
	Define decimal points
		--precision				(3 for mm, for inch by default)
	Add trailing 0s
		--fixed-length				(0 (disabled) by default)
	G0 rapid moves disabled by default, replaced with G1
		--use-rapids				(Set flag to enable G0)
	Disable setting units on output code G20/21
		--omit-units				(Set flag to avoid setting units)
	Force two digit codes G01 insted of G1
		--two-digit-codes			(Set flag to enable)
	Add + sign to positive coordinates
		--force-sign				(Set flag to enable)
	Ignore unsuported operations, use Labels and separate with ','
		--ignore-operations			(Empty by default)

#FreeCAD/6049

### CODE REVIEW

**Comment by luzpaz** [comment]
@alromh87 thank you for the contribution. Would you mind fixing the whitespace in the postprocessor output
Screenshot_20231101_083412

**Comment by luzpaz** [comment]
Please fix typos as well
```
Warning: src/Mod/Path/Path/Post/scripts/wedm_post.py:553: unneded ==> unneeded
Warning: src/Mod/Path/Path/Post/scripts/wedm_post.py:553: aplying ==> applying
Warning: src/Mod/Path/Path/Post/scripts/wedm_post.py:605: writen ==> written
```

**Comment by Syres916** [comment]
This PR is not complete, as src/Mod/Path/CMakeLists.txt requires modifying, as an example see https://github.com/FreeCAD/FreeCAD/commit/b9537089956d01afb5c376a9d8c6d899c99ef71c

**Comment by alromh87** [comment]
@luzpaz I fixed typos and used rebase to change the log message, hope that's the right way to do it.
@Syres916 Added a line to CMakeLists.txt if I understood correctly the PR should be right now

**Comment by luzpaz** [comment]
This PR needs to be rebased. @alromh87 do you know how to do it?

**Comment by alromh87** [comment]
Should I rebase it to latest main?

I think I can do it directly from GitHub web interface.
Otherwise I think it think I can rebase from cli and then force push, please correct me if I'm wrong

**Comment by luzpaz** [comment]
Github interface sucks when it comes to rebasing...unless you use the github codespace functionality. Looks like you figured it out. Thanks!

File: src/Mod/Path/Path/Post/scripts/wedm_post.py [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+                            and (currLocation[param] == c.Parameters[param])
+                       ):
+                            continue
+                        else:
+                            if RELATIVE_GCODE and (param != "I" and param != "J"):
+                                pos =  Units.Quantity(c.Parameters[param] - currLocation[param], FreeCAD.Units.Length)
+                                print(f'currlocation: {currLocation[param]} param: {c.Parameters[param]} pos: {pos}')
+                                if pos == 0:
+                                    # Remove no movement
+                                    continue;
```
Comment (luzpaz) [inline src/Mod/Path/Path/Post/scripts/wedm_post.py:542]: Remove semicolon

File: src/Mod/Path/Path/Post/scripts/wedm_post.py [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+            and hasattr(obj.Base, "CoolantMode")
+        ):
+            if hasattr(obj, "CoolantMode"):
+                coolantMode = obj.CoolantMode
+            else:
+                coolantMode = obj.Base.CoolantMode
+
+        # turn coolant on if required
+        if OUTPUT_COMMENTS:
+            if not coolantMode == "None":
```
Comment (luzpaz) [inline src/Mod/Path/Path/Post/scripts/wedm_post.py:360]: What about `if coolantMode is not "None":`

**Review by luzpaz — CHANGES_REQUESTED** [review]

**Comment by luzpaz** [comment]
There are more linting warnings in https://github.com/FreeCAD/FreeCAD/actions/runs/6724416304/job/18296194303?pr=11248#step:7:93

**Comment by luzpaz** [comment]
No lint fixes ?

**Pull request merged**